### PR TITLE
[Sketcher] Define double click on edges : Select the whole wire.

### PR DESCRIPTION
**Pull request by PaddleStroke**
Fixes https://github.com/FreeCAD/FreeCAD/issues/11605

Double click on an edge select all the connected geometries.

### CODE REVIEW

**Comment by pierreporte** [comment]
Thank you for this.

What was your approach for wires that have geometry connected of a different type (construction or not)?

**Comment by PaddleStroke** [comment]
No distinction. If geometries have their end are coincident, even if without constraints then they are selected.

**Comment by PaddleStroke** [comment]
@adrianinsaval Do you perhaps have some time to review this PR?

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp [hunk 1]
```diff
@@ -1370,6 +1378,70 @@ void ViewProviderSketch::editDoubleClicked()
     }
 }
 
+void ViewProviderSketch::toggleWireSelelection(int clickedGeoId)
+{
+    Sketcher::SketchObject* obj = getSketchObject();
+
+    const Part::Geometry* geo1 = obj->getGeometry(clickedGeoId);
+    if (isPoint(*geo1) || isCircle(*geo1) || isEllipse(*geo1)) {
```
Comment (adrianinsaval) [inline src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:1386]: should this check for closed loop bsplines too or is that overkill?

**Comment by adrianinsaval** [comment]
What do you guys think the behavior should be when there's a fork?
image

As I see it with this implementation which loop gets selected depends on the order of creation of the edges and there's no way for the user to choose which gets selected. I don't have a problem with this behavior as I don't think I would use this functionality that much.

**Comment by pierreporte** [comment]
I would expect the whole thing to be selected, regardless of the starting edge. QCAD works this way.

**Comment by adrianinsaval** [comment]
Nevermind, I misinterpreted the code the first time, All connected edges will be selected.

File: src/Mod/Sketcher/Gui/ViewProviderSketch.cpp (hunk 1 above)
Comment (PaddleStroke) [reply src/Mod/Sketcher/Gui/ViewProviderSketch.cpp:1386]: That would make sense.

**Comment by PaddleStroke** [comment]
@adrianinsaval I have added the condition for periodic bspline

**Comment by pierreporte** [comment]
@PaddleStroke I just found an issue with this: #16198.

**Comment by maxwxyz** [comment]
@PaddleStroke this Sketcher feature is not working anymore, see #16198, can confirm
```
OS: Windows 11 build 22631
Word size of FreeCAD: 64-bit
Version: 1.0.0RC2.38806 (Git)
Build type: Release
Branch: (HEAD detached at 1.0rc2)
Hash: 3d63fc6c2f665a8d5e6468845a419bcac80756c7
Python 3.11.9, Qt 5.15.13, Coin 4.0.3, Vtk 9.2.6, OCC 7.7.2
Locale: German/Germany (de_DE)
Stylesheet/Theme/QtStyle: FreeCAD Dark.qss/FreeCAD Dark/Fusion
Installed mods: 
  * CfdOF 1.27.10
  * Curves 0.6.47
  * dodo 1.0.1
  * fasteners 0.5.29
  * fasteners.backup1721936159.7542229 0.5.24 (Disabled)
  * freecad.gears 1.3.0
  * Manipulator 1.5.7
  * OpenTheme 2024.9.1
  * OpticsWorkbench 1.0.24
  * Rocket 3.3.0
  * sheetmetal 0.4.26
  * Silk 0.1.5
  * Silk.backup1727166062.7227929 0.1.5 (Disabled)

```

**Comment by PaddleStroke** [comment]
I confirm the feature is broken. If you can open an issue about this? https://github.com/FreeCAD/FreeCAD/issues/16198 is closed and was specific to slot but the feature is totally broken apparantly.

**Pull request merged**